integration/containerd_image_test.go

```diff
@@ -237,6 +237,39 @@ func TestContainerdSandboxImage(t *testing.T) {
 	assert.True(t, pimg.Pinned)
 }
 
+func TestContainerdSandboxImagePulledOutsideCRI(t *testing.T) {
+	var pauseImage = images.Get(images.Pause)
+	ctx := context.Background()
+
+	t.Log("make sure the pause image does not exist")
+	imageService.RemoveImage(&runtime.ImageSpec{Image: pauseImage})
+
+	t.Log("pull pause image")
+	_, err := containerdClient.Pull(ctx, pauseImage)
+	assert.NoError(t, err)
+
+	t.Log("pause image should be seen by cri plugin")
+	var pimg *runtime.Image
+	for retry := 0; retry < 10; retry++ {
+		pimg, err = imageService.ImageStatus(&runtime.ImageSpec{Image: pauseImage})
+		if err == nil && pimg != nil {
+			break
+		}
+		time.Sleep(time.Second)
```

Comment:
you can use `eventually` here 
https://github.com/containerd/containerd/blob/96bf529cbf55940ddb96bb8adc8be51b11922ebb/integration/containerd_image_test.go#L92

Reply:
hmm... is a good way to find out if we have some delay/deadlock/parallel situations on the same cache.. 10tries 10seconds is a long enough time maybe?